core/src/main/clojure/xtdb/xtql.clj

```diff
@@ -126,13 +132,20 @@
   (plan-expr [this] (param-sym (subs (.v this) 1)))
   (required-vars [_this] #{})
 
+  Expr$Vec
+  (plan-expr [this] (into [] (map plan-expr) (.elements this)))
+  (required-vars [this] (into #{} (mapv required-vars (.elements this))))
```

Comment:
needs to be mapcat in required vars as each child element will also return a collection